### DISCUSSION TITLE
Suppliers & Brands names are not multilingual

### DISCUSSION THREAD

**Discussion by mohammed-alsiddeeq** [PrestaShop/PrestaShop#40199] [Ideas]
I see that the suppliers and the brands names are not multilingual
I am more than ready to help make any pull request if needed

**Comment by matks** [comment]
Hi @mosid thank you for offering help !

By "translatable" you mean "being able to define a name for each enabled language ?

We already considered this idea, but it creates a lot of issues for people upgrading their shops.

Let's say we introduce this change for PS 1.7.8.0 If we introduce translatable supplier names, it works well for people who create a new 1.7.8 shop and setup their database for the 1st time. But for people upgrading from 1.7.7. to 1.7.8 they have to link all database fields previously targeting a single row to now multiple rows.

This is manageable through a heavy SQL processing for upgrades (although a bit risky 😅) but this is very complex to manage for installed modules. Any module the previously single field needs to update to handle the many fields that now exist.

That is why this feature might not be done for PS 1.7 ... to preserve existing modules ecosystem from huge breaking changes 😢

**Comment by khouloudbelguith** [comment]
Hi @mosid,

Thanks for your report.
I have the same issue with PS1.7.5.2, PS1.7.6.4, and the branch 1.7.7.x.
![image](https://user-images.githubusercontent.com/16067358/78369586-b42bd500-75c5-11ea-91f5-1b0effd7baa6.png)
![image](https://user-images.githubusercontent.com/16067358/78369606-bee66a00-75c5-11ea-9e83-ee86ee8bc50e.png)

@matks, we have an issue similar but in the carriers page: 
https://github.com/PrestaShop/PrestaShop/issues/12901

Ping @sam-pires @PrestaShop/prestashop-product-team what do you think of this improvement?

Thanks!

**Comment by LouiseBonnard** [comment]
Hello ! Following what @matks said, I put this issue in the Next Major repository so that it can be studied as soon as a new major version is on its way. ;-)

**Comment by Eolia** [comment]
A suplier name or brand name should not be translatable, it's a name.... Do you translate your own name ???
Apple, Microsoft, PrestaShop...

**Comment by marsaldev** [comment]
@Eolia 
There are some cases where the same brand are called with different names, like "Dr.Oekter" in Germany, but in Italy is "Cameo"

https://images.app.goo.gl/S7VaQwMuGmHusEvu5

At this point I suggest to give the possibility to make also the logo 'multilanguage' 😊

Maybe there are many others cases that we doesn't not aware of 😊

**Comment by mohammed-alsiddeeq** [comment]
> A suplier name or brand name should not be translatable, it's a name.... Do you translate your own name ???
> Apple, Microsoft, PrestaShop...

I mean showing it in other languages
e.g: Sony is , سوني in arabic
I hope it is clear

**Comment by marsaldev** [comment]
@mosid Another interesting prospective 😉

**Comment by andromaque** [comment]
> Let's say we introduce this change for PS 1.7.8.0 If we introduce translatable supplier names, it works well for people who create a new 1.7.8 shop and setup their database for the 1st time. But for people upgrading from 1.7.7. to 1.7.8 they have to link all database fields previously targeting a single row to now multiple rows.
> 
> This is manageable through a heavy SQL processing for upgrades (although a bit risky 😅) but this is very complex to manage for installed modules. Any module the previously single field needs to update to handle the many fields that now exist.

@matks For the transition period, the field "name" could stay in the database (but not referenced anymore in the BO). That way, we do not break any module, and for the next major this field could be removed.
Also I don't think that the upgrade would be a heavy SQL script, the current name could be simply copied for every language activated, and modified later by the shop owner in the BO.

For the reference, brands in Greece are also often written in greek for the greek market, but written in latin for the rest of the world, and the same for the logo. E.g. ΠΑΠΑΔΟΠΟΥΛΟΥ / PAPADOPOULOS

**Comment by matks** [comment]
> @matks For the transition period, the field "name" could stay in the database (but not referenced anymore in the BO). That way, we do not break any module, and for the next major this field could be removed.

This strategy could indeed provide some backward compatibility if we keep the old single-lang field and the new multi-lang field synchronized. Keeping the field but stop using it does not work: module rely on a field and its content 😅 .

But this is an additional overhead of code+logic. We will need to keep the "english" name duplicated in the old field and the new field, make sure it is updated in both places whether the update happens through BO, through webservice, through CSV import.

**Comment by andromaque** [comment]
@matks Of course we have to keep it synchronized 😄 A trigger could do the trick though? That way, there is no need for additional code in the BO that could potentially lead to new bugs.
